doc/oper.xml

```diff
@@ -2131,3 +2133,55 @@ gap> data;
 </Description>
 </ManSection>
 <#/GAPDoc>
+
+<#GAPDoc Label="IsDigraphPath">
+<ManSection>
+  <Oper Name="IsDigraphPath" Arg="D, v, a"/>
+  <Oper Name="IsDigraphPath" Arg="D, list" Label="for a digraph and list"/>
+  <Returns><K>true</K> or <K>false</K>.</Returns>
+  <Description>
+    This function returns <K>true</K> if the arguments <A>v</A> and <A>a </A>
+    describe a path in the digraph <A>D</A>.
+
+    A directed path (or directed cycle) of non-zero length <C>n-1</C>,
```

Comment:
I don't like the duplication of this block of text just for reasons of maintainance.

I'd say that the practical definition of should be given here, as it is, and that the documentation for `DigraphPath` should point here. So something like:
```
This function returns <K>true</K> if the arguments <A>v</A> and <A>a</A> describe a path in the digraph <A>D</A>.

The mathematical definition of a directed path and a directed cycle is given
in Section <Ref Subsect="Definitions" Style="Text"/>.
<P/>

In practice, a directed path (or directed cycle) of non-zero length <C>n-1</C>...
```

Reply:
Also good

Reply:
I'll try to make the changes today and re-push.

Reply:
I've only got back to this today, and I think I've made all the changes you requested @wilfwilson, and although I thought I understood your comment above (the one "In practice", that I'm replying to) but now I don't. What would follow "In practice" precisely?

Reply:
Hey @james-d-mitchell, happy to try to clarify what I meant, and the way that I think about things.

The text at `<Ref Subsect="Definitions" Style="Text"/>` in the introduction contains (amongst other definitions) the following sentences:
> A directed walk on a digraph is a sequence of alternating vertices and edges (v_1, e_1, v_2, e_2, ..., e_n-1, v_n) such that each edge e_i has source v_i and range v_i+1. A directed path is a directed walk where no vertex (and hence no edge) is repeated

For me, this is the 'mathematical definition'.

However, in a practical sense in GAP/Digraphs, there is no such object as a 'sequence'; a vertex is a positive integer, and an edge is a list of two positive integers. So even if you were happy to consider a sequence as being another name for a list, then the literal interpretation of the mathematical definition would be that an object representing a digraph path in the Digraphs package would be something like `[5, [5,3], 3, [3,2], 2, [2, 6], 6]`. But that's not how we've decided to represent paths in the Digraphs package, we do in differently (and more sensibly). So in that sense, we have a practical definition that is separate from the mathematical definition.

The practical definition is what is (currently) given in the manual entry for `DigraphPath`. i.e. it's a pair of lists, the first one giving the list of vertices that occur, the second one giving the positions of subsequent vertices in the out-neighbours of the previous.

In my opinion, we should only give this 'practical definition' once (since it's easier to maintain one copy of a fiddly definition in the documentation); but instead of it being in the manual entry for `DigraphPath`, as it is now, I think it should instead be in the manual entry for `IsDigraphPath`. Then the manual entry for `DigraphPath` should no longer contain the full definition, but it should instead have a reference to the manual entry for `IsDigraphPath` in its place.

To make it more concrete:
```
< Manual entry for IsDigraphPath >

This function returns <K>true</K> if the arguments <A>v</A> and <A>a</A>
describe a path in the digraph <A>D</A>.
<P/>

The mathematical definition of a directed path and a directed cycle is given
in Section <Ref Subsect="Definitions" Style="Text"/>.
<P/>

In practice, a directed path (or directed cycle) of non-zero length
<C>n-1</C> (v_1, e_1, v_2, e_2, ..., e_n-1, v_n),
is represented by a pair of lists [v,a] as follows:
* v is the list [v_1, v_2, ..., v_n].
* ... [continue with the rest of the text from the manual entry for `DigraphPath`]
```
and then the manual entry for `DigraphPath` should look something like:
```
Returns:  A pair of lists, or fail.

  If  there  exists  a non-trivial directed path (or a non-trivial cycle, in the case that u = v) from
  vertex  u  to  vertex v in the digraph digraph, then this operation returns such a directed path (or
  directed cycle). Otherwise, this operation returns fail.

  See Section 'Definitions' for the mathematical definition of a directed path and a directed cycle,
  and see IsDigraphPath for the way in which paths and cycles are represented in the Digraphs package.
```

Reply:
In short, what should follow "In practice," is the paragraph-bulletpoint-bulletpoint-paragraph that currently forms the second half of the manual entry for `DigraphPath`.

Reply:
Sorry @james-d-mitchell I didn't do a great job of this 😬. Having looked at the diff and remembering the context more, I think I can say it better:

I am suggesting that you add:
```
<P/>

The mathematical definition of a directed path and a directed cycle
is given in Section <Ref Subsect="Definitions" Style="Text"/>.
<P/>

In practice,
```
between the first and second paragraphs that you've already included in this PR.

And also I'm suggesting that you shorten the documentation for `DigraphPath`, by pointing to `IsDigraphPath` rather than repeat the same information.